src/api/http.rs

```diff
@@ -219,17 +210,29 @@ fn partition_by_id(
         })
 }
 
-#[put("/vdisks/<vdisk_id>/partitions/<partition_id>/<action>")]
+#[get("/vdisks/<vdisk_id>/partitions/<partition_id>/<action>")]
 fn change_partition_state(
-    _bob: State<BobSrv>,
-    vdisk_id: usize,
-    partition_id: usize,
+    bob: State<BobSrv>,
+    vdisk_id: u32,
+    partition_id: i64,
     action: Action,
-) -> String {
-    format!(
-        "{:?} partittion {} of vd {}",
-        action, partition_id, vdisk_id
-    )
+) -> Result<StatusExt, StatusExt> {
+    let group = find_group(&bob, vdisk_id)?;
+    let group = group.clone();
+    let mut rt = Runtime::new().expect("create runtime");
+    let res = format!(
+        "partition with id: {} in vdisk {} is successfully {:?}ed",
+        partition_id, vdisk_id, action
+    );
+    let task = async move {
+        match action {
+            Action::Attach => group.attach(partition_id).await,
+            Action::Detach => group.detach(partition_id).await,
+        }
+    };
+    rt.block_on(task);
```

Comment:
why `block_on` here? can we go with `await`?

Reply:
Api server runs in its own thread. And runtime is required for only `change_partition_state`, for now. So in this case it is easier to block on a task than to solve problems with launching rocket server with our own tokio runtime.